source/fe/fe_raviart_thomas.cc

```diff
@@ -248,6 +248,173 @@ FE_RaviartThomas<dim>::initialize_support_points(const unsigned int deg)
+  // same number of dofs
+  AssertDimension(this->n_unique_faces(), 1);
+  const unsigned int face_no = 0;
+
+  Assert(this->adjust_quad_dof_index_for_face_orientation_table[0]
+             .n_elements() == 8 * this->n_dofs_per_quad(face_no),
+         ExcInternalError());
+
+  // The 3D RaviartThomas space has tensor_degree*tensor_degree face dofs
+  const unsigned int n = this->tensor_degree();
+  Assert(n * n == this->n_dofs_per_quad(face_no), ExcInternalError());
+
+  // The vector of tables adjust_quad_dof_index_for_face_orientation_table
+  // contains offsets for local face_dofs and is being filled with zeros in
+  // fe.cc. We need to fill it with the correct values in case of non-standard,
+  // flipped (rotated by +180 degrees) or rotated (rotated by +90 degrees) faces
+
+  // The dofs on a face are connected to a n x n matrix. for example, for
+  // tensor_degree==3 we have the following dofs on a quad:
+
+  //  ___________
+  // |           |
+  // |  6  7  8  |
+  // |           |
+  // |  3  4  5  |
+  // |           |
+  // |  0  1  2  |
+  // |___________|
+  //
+  // We have dof_index=i+n*j with index i in x-direction and index j in
+  // y-direction running from 0 to n-1.  to extract i and j we can use
+  // i=dof_index%n and j=dof_index/n. The indices i and j can then be used to
+  // compute the offset.
+
+  // Example: if the switches are (true | true | true) that means we flip the
+  // face first by + 90 degree(counterclockwise) then by another +180
+  // degrees but we do not flip it since the face has standard
+  // orientation. The flip axis is the diagonal from the lower left to the upper
+  // right corner of the face. With these flags the configuration above becomes:
+  //  ___________
+  // |           |
+  // |  8  7  6  |
+  // |           |
+  // |  5  4  3  |
+  // |           |
+  // |  2  1  0  |
+  // |___________|
+  //
+  // This is exactly what is realized by the formulas implemented below. Note
+  // that the necessity of a permuattion depends on the three flags.
+
+  // There is also a pattern for the sign change of the mapped face_dofs
+  // depending on the switches. In the above example it would be
+  //  ___________
+  // |           |
+  // |  +  +  -  |
+  // |           |
+  // |  +  +  -  |
+  // |           |
+  // |  +  +  -  |
+  // |___________|
+  //
+
+  for (unsigned int local_face_dof = 0;
+       local_face_dof < this->n_dofs_per_quad(face_no);
+       ++local_face_dof)
+    {
+      // Row and column
+      unsigned int i = local_face_dof % n, j = local_face_dof / n;
+
+      // We have 8 cases that are all treated the same way. Note that the
+      // corresponding case to case_no is just its binary representation.
+      // The above example of (false | true | true) would be case_no=3
+      for (unsigned int case_no = 0; case_no < 8; ++case_no)
+        {
+          // Get the binary representation of the case
+          const bool face_orientation = (((case_no / 4) % 2) == 1);
```

Comment:
`Utilities::get_bit(case_no, 2)` should do this for you :-)

Reply:
Thanks, changed that :-)